app/main.py

```diff
@@ -1,3 +1,58 @@
+from typing import Union
+from functools import total_ordering
+
+
+@total_ordering
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: Union[int, float]) -> None:
+        self.km = km
```

Comment:
For the Distance class, it would be a good practice to add a check to ensure that the distance (km) being passed to the initializer is not negative. Distances are typically non-negative in most contexts.